scalafix/rules/src/main/scala/fix/0.8/MigrateV0_8.scala

```diff
@@ -0,0 +1,143 @@
+package fix
+package v0_8_0
+
+import scalafix.v1._
+import scala.meta._
+
+private class FixRunWithContext extends SemanticRule("FixRunWithContext") {
+
+  private def fixSubtree(name: String)(doc: SyntacticDocument): Patch = {
+    doc.tree.collect {
+      case t @ Term.Select(Term.Name(`name`), Term.Name("isCompleted")) =>
+        Patch.empty
+      case t @ Term.Select(Term.Name(`name`), Term.Name("state")) =>
+        Patch.empty
+      case t @ Term.Select(Term.Name(`name`), x) =>
+        Patch.replaceTree(t, s"$name.waitUntilFinish.$x")
+      case t @ Term.Apply(tn, ps) if ps.find(x => x.toString == name).isDefined =>
+        val ps2 =
+          ps.map {
+            case t if t.toString == name => q"$t.waitUntilFinish"
+            case t => t
+          }
+        Patch.replaceTree(t, Term.Apply(tn, ps2).toString)
+    }.asPatch
+  }
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    doc.tree.collect {
+      // Convert val`x` of type ScioExecutionContext to ScioResult
+      // when `x` is passed to a method oo
+      // if a method is called on `x` (except `isCompleted and `state`)
+      case t @ q"val ${x} = runWithContext(${body})" =>
+        val children = t.parent.toList.flatMap(_.children).filterNot(_ == t)
+        val name = x.toString()
+        children.map { c =>
+          fixSubtree(name)(SyntacticDocument.fromTree(c))
+        }.asPatch
+      // Convert ScioExecutionContext to ScioResult in methods that return a ScioResult
+      case t @ q"def $fn(..$ps): ScioResult = {$body}" =>
+        Patch.addRight(t, ".waitUntilFinish")
+      case t =>
+        Patch.empty
+    }.asPatch
+  }
+}
+
+private class FixScioIO extends SemanticRule("FixScioIO") {
+
+  // Check that the method is a member of an implementation of ScioIO
+  private def isScioIOMember(t: Tree)(implicit doc: SemanticDocument) =
+    t.parent.collect {
+      case p @ Template(_) =>
+        p.inits.exists(_.tpe.symbol == Symbol("com/spotify/scio/io/ScioIO#"))
+    }.getOrElse(false)
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
```

Comment:
Correct me if I am wrong, but I don't think this is modifying imports.

I saw this change of going from `Future` to `ClosedTap`. Doesn't this need a change in the import as well?

[This](https://ghe.spotify.net/payments/pipeline-subs-platform/pull/1168/files#diff-31ada5de4461a63e507abdd6ce0b34fb) is what I was doing manually.